### PR TITLE
Control value of ConstantPostprocessor

### LINKED ISSUE #24247: Control `default` on `Receiver` postprocessor
## Reason
Sending a value from the stochastic tools app into the `Receiver` postprocessor seems like the most general way to send a scalar value into a MOOSE app (because the value of that postprocessor can then be used in numerous ways). But the `default` value is not controllable.

## Design
Make `default` controllable.

## Impact
New feature.

### PR DESCRIPTION
**Pull request by aprilnovak**
Closes #24247

### CODE REVIEW

**Review by lindsayad — DISMISSED** [review]
Would be nice to have a test, but not requiring it

**Comment by aprilnovak** [comment]
Can do!

**Comment by aprilnovak** [comment]
Good call @lindsayad, it actually wasn't as trivial as I though the one-line would translate to (but that's just a special case for this postprocessor, I think). If you prefer I instead just make a `ConstantPostprocessor` to accomplish this, happy to do that as an alternative.

File: framework/src/postprocessors/Receiver.C [hunk 1]
```diff
@@ -37,18 +39,22 @@ Receiver::Receiver(const InputParameters & params)
...
 {
-  // Return the stored value (references stored value in getPostprocessorData)
-  return _my_value;
+  // if the current value is the same as the default, then we assume we are changing
+  // value according to an external actor. Otherwise, we assume we are changing
+  // value according to the default (which is controllable)
+  if (MooseUtils::absoluteFuzzyEqual(_default, _first_value))
+    return _my_value;
+  else
+    return _default;
```
Comment (lindsayad) [inline framework/src/postprocessors/Receiver.C:59]: Hmm this just doesn't feel right to me. Someone more familiar with reporters, @loganharbour @zachmprince, how would you guys go about what @aprilnovak wants to do in #24247 ?

**Review by lindsayad — CHANGES_REQUESTED** [review]

File: framework/src/postprocessors/Receiver.C (hunk 1 above)
Comment (aprilnovak) [reply framework/src/postprocessors/Receiver.C:59]: I just want to control a postprocessor which just holds a value - I do agree this feels weird and could have conflict resolution issues if mistakenly used a `MultiAppPostprocessorTransfer` to write into the postprocessor, while _also_ controlling it. I could do a `ConstantPostprocessor` instead, lmk

Comment (lindsayad) [reply framework/src/postprocessors/Receiver.C:59]: A `ConstantPostprocessor` sounds really good to me

File: test/tests/postprocessors/constant/tests [hunk 2]
```diff
@@ -0,0 +1,10 @@
+[Tests]
+  [receiver]
+    type = CSVDiff
+    input = receiver.i
+    csvdiff = receiver_out.csv
+    requirement = "The system shall allow the value of a ConstantPostprocessor to be controllable"
```
Comment (lindsayad) [inline test/tests/postprocessors/constant/tests:6]:
```suggestion
    requirement = "The system shall allow the value of a postprocessor to be controllable."
```
Requirements should be agnostic of implementation details. In theory, the requirement should come first, and then we implement the capability that accomplishes the requirement

**Review by lindsayad — CHANGES_REQUESTED** [review]
This looks much better. Thank you!

**Review by lindsayad — APPROVED** [review]
Very nice, thank you!

**Comment by lindsayad** [comment]
I've opened https://github.com/idaholab/moose/issues/24285 regarding the NS timeout. It's not related to this PR

**Comment by lindsayad** [comment]
Thank you @aprilnovak !

**Pull request merged**